From fbc0b349edfa56ad0cad7d4a6df7c1b1ada7d5fa Mon Sep 17 00:00:00 2001
From: Kai Uwe Broulik <kde@privat.broulik.de>
Date: Mon, 15 Apr 2024 22:43:15 +0200
Subject: [PATCH] udisks: Return empty string for "root" clearTextPath

There cannot be an empty object path on DBus, so "/" is used to denote
no clear text device. Return empty string explicitly here to avoid
littering even more places with empty || "/" checks.

BUG: 485507
(cherry picked from commit 9ecaf388ccd41d2831ab3833c26627f72ab78f40)
---
 src/solid/devices/backends/udisks2/udisksstorageaccess.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/solid/devices/backends/udisks2/udisksstorageaccess.cpp b/src/solid/devices/backends/udisks2/udisksstorageaccess.cpp
index 9a77cc40..3662c4c7 100644
--- a/src/solid/devices/backends/udisks2/udisksstorageaccess.cpp
+++ b/src/solid/devices/backends/udisks2/udisksstorageaccess.cpp
@@ -363,7 +363,11 @@ QString StorageAccess::generateReturnObjectPath()
 
 QString StorageAccess::clearTextPath() const
 {
-    return m_device->prop("CleartextDevice").value<QDBusObjectPath>().path();
+    const QString path = m_device->prop("CleartextDevice").value<QDBusObjectPath>().path();
+    if (path != QLatin1String("/")) {
+        return path;
+    }
+    return QString();
 }
 
 bool StorageAccess::requestPassphrase()
-- 
2.44.0

